perun-core/src/main/java/cz/metacentrum/perun/taskslib/dao/jdbc/TaskResultDaoJdbc.java

```diff
@@ -293,20 +292,52 @@ public int clearOld(int engineID, int numDays) {
 				dataOut.write(data[i]);
 		}
 		if(maxLength < data.length) {
-			// we had to cut the byte array at limit
-			// data[limit-1] is the last added byte
-			// we have to check, if it starts the non-ASCII char sequence
-			if(data[limit-1] >= 0xC0 ) {
-				dataOut.write(data[limit]);
-			}
-			if(data[limit-1] >= 0xE0 ) {
-				dataOut.write(data[limit+1]);
-			}
-			if(data[limit-1] >= 0xF0 ) {
-				dataOut.write(data[limit+2]);
+			// check if we have to add some bytes in case we have split an UTF-8 character that is longer than 1 byte.
+			if (!isASingleByteUTF8Char(data[limit-1])) {
+				int i = 0;
+				while (!isAStartingByteUTF8Char(data[limit + i])) {
+					dataOut.write(data[limit + i]);
+					i++;
+				}
```

Comment:
I would add some testing of the maximal length of UTF8 character. It will prevent problems with wrong formatted input. What do you think?

Reply:
That's a good point. I will add it.

Reply:
Done.